.github/workflows/benches.yml

```diff
@@ -0,0 +1,35 @@
+name: Benchmark Suite
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+        - master
+
+env:
+  VERBOSE: true
+  ENVIRONMENT_UPSTREAM: docker.pkg.github.com/timberio/vector/environment:${{ github.sha }}
+
+jobs:
+  cancel-previous:
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.3.1
+        with:
+          workflow_id: 1481114
+          access_token: ${{ secrets.GITHUB_TOKEN }}
+
+
+  bench:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v10
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - run: nix-shell --pure --run "echo 'Env installing...'"
```

Comment:
What if we set [`shell`](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun) to `nix-shell --pure --run` for the whole job? If it works, it'll save us a few characters on every command. :smile: Maybe we can even set is as [default](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun).

Reply:
I was looking for that option! Thanks!

Reply:
So one note, we can't use `nix-shell` until we've gotten cachix. :(

Reply:
![image](https://user-images.githubusercontent.com/130903/84052198-178b0800-a965-11ea-815a-7ee4876856ba.png)

Reply:
Oh, unexpected. Here's the relevant code: https://github.com/actions/runner/blob/2bd0b1af0e4daa0ed3d8115c79ba6aa6633dd732/src/Runner.Worker/Handlers/ScriptHandler.cs#L220
It might be helpful if you want to dig into that - I've spent enough time with Github Actions to know checking the code directly is quicker than trying to find something like this in the docs.

Reply:
> So one note, we can't use `nix-shell` until we've gotten cachix. :(

Don't we use `cachix` by default?

Reply:
The first step of most CI jobs now is installing nix, which can't be done in nix-shell since we don't have one. :)

Reply:
Ahhh ok I have an idea then

Reply:
We can just set `shell: bash` specifically for those first steps

Reply:
I.e., `shell: nix` at the `job` level (on the contrary to `default` - to not spill the default to newly added jobs that don't expect shell to be nix), and `shell: bash` for the setup step.

Reply:
Also, I think quotes around `{0}` aren't needed